src/main/scala/beam/utils/plan/sampling/AvailableModeUtils.scala

```diff
@@ -36,16 +37,15 @@ object AvailableModeUtils extends LazyLogging {
     * @return List of excluded mode string
     */
   def getExcludedModesForPerson(population: Population, personId: String): Array[String] = {
-    Option(
+    val maybeExcludedModes = Option(
       population.getPersonAttributes.getAttribute(personId, PopulationAdjustment.EXCLUDED_MODES)
-    ).map(_.toString) match {
-      case Some(modes) =>
-        if (modes.isEmpty) {
-          Array.empty[String]
-        } else {
-          modes.split(",")
-        }
-      case None => Array.empty[String]
+    )
+    maybeExcludedModes.map(_.toString)
+    maybeExcludedModes match {
+      case Some(modes: Array[String]) => modes.filterNot(isBlank)
+      case Some(modes: Iterable[_])   => modes.flatMap(_.toString.split(",")).filterNot(isBlank).toArray
+      case Some(modes)                => modes.toString.split(",").filterNot(isBlank)
```

Comment:
After adding the scenario to handle strings it is now failing on 

> 20:06:10 ParkingSpec:
> 20:06:10 Parking system
...
> 20:06:10 - must very expensive parking should reduce driving *** FAILED *** (2 minutes, 5 seconds)
> 20:06:10   8 was not greater than 20 (ParkingSpec.scala:225)
> 20:06:10 - must no parking stalls should reduce driving *** FAILED *** (1 minute, 34 seconds)
> 20:06:10   8 was not greater than 20 (ParkingSpec.scala:239)

I'm wondering if this splitting of string (which in some parts was new IIRC) results in test breakage that was never correct?

Reply:
Although, I just re-ran locally and it worked...I am going to kick off again and see if it is a flaky test...I think I may have seen this earlier also...

Reply:
Yah - Already created as #2957